### PR TITLE
Duka location

### PR DESCRIPTION
**Pull request by Bahatiroben**
A client or prospective client can access the duka location functionality fro Kenya
- on the splash menu, clients farmers are required to enter their account numbers and non-client farmers are required to enter zero

- they are presented with their menu including locating an oaf duka
- once they choose it, they are directed into steps to locate the duka and contact the duka supervisor
- duka supervisors are notified once a farmer contacts them.
- farmers receive a message with the supervisor's information

Dev info
-  you can use this account `10477597` to test on [telerivet](https://telerivet.com/p/0c6396c9/service/SVbc04709b5abb48c0)

```
finishes SI-465
finishes SI-464 
finishes SI-463

### CODE REVIEW

File: duka-locator/inputHandlers/contactDukaAgentHandler.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    beforeEach(() => {
+        jest.resetModules();
+    });
+
+    it('should contact the duka agent', () => {
+        contact.phone_number = '0777664839';
+        state.vars.duka_options = JSON.stringify({reach_out_to_agent: 1, exit_menu: 2});
+        state.vars.selected_duka = JSON.stringify({duka_supervisor_pn: '0767443653', duka_supervisor_name: 'Claude'});
+        contactDukaAgentHandler('1');
+        expect(sendMessage).toHaveBeenCalledTimes(2);
```
Comment (nyabongo) [inline duka-locator/inputHandlers/contactDukaAgentHandler.test.js:16]: please add what sendMessage was called with here

File: duka-locator/inputHandlers/countyHandler.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                next: () => {
+                    current= current + 1;
+                    return countiesMock[current];
+                },
+                hasNext: () => current < countiesMock.length -1
+            })
+        };
+        jest.spyOn(project, 'getOrCreateDataTable').mockReturnValue(dukaCountiesTableMock);
+        countyHandler('2');
+        expect(sayText).toHaveBeenCalled();
```
Comment (nyabongo) [inline duka-locator/inputHandlers/countyHandler.test.js:27]: Again here as before, please test what it was called with

File: duka-locator/inputHandlers/countyHandler.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        state.vars.labeled_counties = JSON.stringify({'2': 1});
+        var dukaCountiesTableMock = {
+            num_rows: 1,
+            queryRows: () => ({
+                hasNext: () => false
+            })
+        };
+        jest.spyOn(project, 'getOrCreateDataTable').mockReturnValue(dukaCountiesTableMock);
+        countyHandler('2');
+        expect(sayText).toHaveBeenCalled();
```
Comment (nyabongo) [inline duka-locator/inputHandlers/countyHandler.test.js:43]: This too should assert that sayText  was calledwith a specific string

Comment (Bahatiroben) [reply duka-locator/inputHandlers/countyHandler.test.js:43]: alright. let me go through them.

File: ke-legacy/utils/buildMenu.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const menuBuilder = require('./build-menu');
+
+describe('Build menu', () => {
+    it('should return an array of menus when for many screens for characters > 140', () => {
+        const menu = menuBuilder([{key: 'find_oaf_contact', options: {'$label': 1}},{key: 'find_oaf_contact', options: {'$label': 1}},
```
Comment (nyabongo) [inline ke-legacy/utils/buildMenu.test.js:5]: This test does not seem to test what it says it does

Comment (Bahatiroben) [reply ke-legacy/utils/buildMenu.test.js:5]: Could you shade more light on this? but I guess expecting more line the array length, or the array itself would do the job.

**Pull request merged**